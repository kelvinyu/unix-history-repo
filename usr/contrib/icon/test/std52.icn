record array(a,b,c,d,e,f,g)

procedure dummy(u,v,x,y,z)
   suspend u | v
   return x
end

procedure main()
   write("every write(center(\"abcdef\",1 to 20,\" \" | \"0\" | \"=-\")) ----> ",image(every write(center("abcdef",1 to 20," " | "0" | "=-"))) | "none")
   write("every write(left(\"abcdef\",1 to 20,\" \" | \"0\" | \"=-\")) ----> ",image(every write(left("abcdef",1 to 20," " | "0" | "=-"))) | "none")
   write("every write(right(\"abcdef\",1 to 20,\" \" | \"0\" | \"=-\")) ----> ",image(every write(right("abcdef",1 to 20," " | "0" | "=-"))) | "none")
   write("center(\"\",20,repl(\"x.\",30)) ----> ",image(center("",20,repl("x.",30))) | "none")
   write("left(\"\",20,repl(\"x.\",30)) ----> ",image(left("",20,repl("x.",30))) | "none")
   write("right(\"\",20,repl(\"x.\",30)) ----> ",image(right("",20,repl("x.",30))) | "none")
   write("right(\"\",\"\") ----> ",image(right("","")) | "none")
end
