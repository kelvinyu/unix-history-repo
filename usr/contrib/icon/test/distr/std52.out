d
d
d
cd
cd
cd
cde
cde
cde
bcde
bcde
bcde
bcdef
bcdef
bcdef
abcdef
abcdef
abcdef
abcdef 
abcdef0
abcdef-
 abcdef 
0abcdef0
=abcdef-
 abcdef  
0abcdef00
=abcdef=-
  abcdef  
00abcdef00
=-abcdef=-
  abcdef   
00abcdef000
=-abcdef-=-
   abcdef   
000abcdef000
=-=abcdef-=-
   abcdef    
000abcdef0000
=-=abcdef=-=-
    abcdef    
0000abcdef0000
=-=-abcdef=-=-
    abcdef     
0000abcdef00000
=-=-abcdef-=-=-
     abcdef     
00000abcdef00000
=-=-=abcdef-=-=-
     abcdef      
00000abcdef000000
=-=-=abcdef=-=-=-
      abcdef      
000000abcdef000000
=-=-=-abcdef=-=-=-
      abcdef       
000000abcdef0000000
=-=-=-abcdef-=-=-=-
       abcdef       
0000000abcdef0000000
=-=-=-=abcdef-=-=-=-
every write(center("abcdef",1 to 20," " | "0" | "=-")) ----> none
a
a
a
ab
ab
ab
abc
abc
abc
abcd
abcd
abcd
abcde
abcde
abcde
abcdef
abcdef
abcdef
abcdef 
abcdef0
abcdef-
abcdef  
abcdef00
abcdef=-
abcdef   
abcdef000
abcdef-=-
abcdef    
abcdef0000
abcdef=-=-
abcdef     
abcdef00000
abcdef-=-=-
abcdef      
abcdef000000
abcdef=-=-=-
abcdef       
abcdef0000000
abcdef-=-=-=-
abcdef        
abcdef00000000
abcdef=-=-=-=-
abcdef         
abcdef000000000
abcdef-=-=-=-=-
abcdef          
abcdef0000000000
abcdef=-=-=-=-=-
abcdef           
abcdef00000000000
abcdef-=-=-=-=-=-
abcdef            
abcdef000000000000
abcdef=-=-=-=-=-=-
abcdef             
abcdef0000000000000
abcdef-=-=-=-=-=-=-
abcdef              
abcdef00000000000000
abcdef=-=-=-=-=-=-=-
every write(left("abcdef",1 to 20," " | "0" | "=-")) ----> none
f
f
f
ef
ef
ef
def
def
def
cdef
cdef
cdef
bcdef
bcdef
bcdef
abcdef
abcdef
abcdef
 abcdef
0abcdef
=abcdef
  abcdef
00abcdef
=-abcdef
   abcdef
000abcdef
=-=abcdef
    abcdef
0000abcdef
=-=-abcdef
     abcdef
00000abcdef
=-=-=abcdef
      abcdef
000000abcdef
=-=-=-abcdef
       abcdef
0000000abcdef
=-=-=-=abcdef
        abcdef
00000000abcdef
=-=-=-=-abcdef
         abcdef
000000000abcdef
=-=-=-=-=abcdef
          abcdef
0000000000abcdef
=-=-=-=-=-abcdef
           abcdef
00000000000abcdef
=-=-=-=-=-=abcdef
            abcdef
000000000000abcdef
=-=-=-=-=-=-abcdef
             abcdef
0000000000000abcdef
=-=-=-=-=-=-=abcdef
              abcdef
00000000000000abcdef
=-=-=-=-=-=-=-abcdef
every write(right("abcdef",1 to 20," " | "0" | "=-")) ----> none
center("",20,repl("x.",30)) ----> "x.x.x.x.x.x.x.x.x.x."
left("",20,repl("x.",30)) ----> "x.x.x.x.x.x.x.x.x.x."
right("",20,repl("x.",30)) ----> "x.x.x.x.x.x.x.x.x.x."
Run-time error 101 at line 15 in std52.icn
integer expected
offending value: ""
